compute/cker/include/cker/train/operation/BinaryArithmetic.h

```diff
@@ -51,6 +54,15 @@ void BinaryArithmeticGrad(const Shape &incoming_shape, const T *incoming_data,
     break;
 
     case ArithmeticType::kSub:
+    {
+      BroadcastTo(incoming_shape, const_cast<T *>(incoming_data), lhs_grad_shape, lhs_grad_data);
+
+      auto const in_map = MapAsMatrixWithLastDimAsRows(incoming_data, incoming_shape);
+      auto rhs_grad_map = MapAsMatrixWithLastDimAsRows(rhs_grad_data, rhs_grad_shape);
+      rhs_grad_map = -in_map;
```

Comment:
(Optional)
If you are not going to support broadcast in this kernel, this code may be replaced with as below.
```suggestion
      auto const in_map = MapAsVector(incoming_data, incoming_shape);
      auto rhs_grad_map = MapAsVector(rhs_grad_data, rhs_grad_shape);
      rhs_grad_map.array() = -in_map.array();
```
I'm not sure, but vectors may be slightly faster than matrixes.

Reply:
Updated

Reply:
(not for this PR) 

https://github.com/Samsung/ONE/blob/8185cb17da4f1e6d84bb14758fec693dd9c38c62/compute/cker/include/cker/eigen/Utils.h#L37-L47

Hmm.. It looks like `MapAsVector` just creates another matrix .. `Eigen::Matrix<Scaler, N, 1>`.
We might need to introduce Util which directly creates `Eigen::Aray()` instead of creating `Matrix`.